WalletWasabi.Fluent/Extensions/TimeSpanMixin.cs

```diff
@@ -0,0 +1,48 @@
+namespace WalletWasabi.Fluent.Extensions;
+
+public static class TimeSpanMixin
+{
+	public static TimeSpan Round(this TimeSpan input)
+	{
+		if (input.Days > 0)
```

Comment:
What about implementing this same thing but with something like this:

```c#
static TimeSpan RoundUp(TimeSpan ts, int minutes) =>
    TimeSpan.FromMinutes(minutes * Math.Ceiling(ts.TotalMinutes / minutes);
```
Or make it an extension method. Then you can use it as follow:
```c#
return input
   .Round(24 * 60)
   .Round(12 * 60)
   .Round(01 * 60);
```

You can see working here (f# sorry, it is much easier to try things):

```f#
> let roundUp (ts:TimeSpan) (mins:int) =                                       
-   TimeSpan.FromMinutes((float)mins * Math.Ceiling(ts.TotalMinutes / (float)mins));;
val roundUp: ts: TimeSpan -> mins: int -> TimeSpan

> let ts = TimeSpan.FromMinutes(25);;
val ts: TimeSpan = 00:25:00

> roundUp ts 30;;
val it: TimeSpan = 00:30:00 {Days = 0;
                             Hours = 0;
                             Milliseconds = 0;
                             Minutes = 30;
                             Seconds = 0;
                             Ticks = 18000000000L;
                             TotalDays = 0.02083333333;
                             TotalHours = 0.5;
                             TotalMilliseconds = 1800000.0;
                             TotalMinutes = 30.0;
                             TotalSeconds = 1800.0;}

> let ts = TimeSpan(0, 13, 0, 0)                                                     
- ;;
val ts: TimeSpan = 13:00:00

> roundUp ts (60 * 24);;             
val it: TimeSpan = 1.00:00:00 {Days = 1;
                               Hours = 0;
                               Milliseconds = 0;
                               Minutes = 0;
                               Seconds = 0;
                               Ticks = 864000000000L;
                               TotalDays = 1.0;
                               TotalHours = 24.0;
                               TotalMilliseconds = 86400000.0;
                               TotalMinutes = 1440.0;
                               TotalSeconds = 86400.0;}

```

Reply:
Oh, I don't know F#, but it would be fine as long as tests pass! 
Also, please notice that the Round method doesn't simply round minutes, but hours!

These are the tests
https://github.com/SuperJMN/WalletWasabi/blob/time-estimation/WalletWasabi.Tests/UnitTests/Extensions/TimeSpanMixinTests.cs

Reply:
Ok. I tried but the logic is hard to understand from a user point of view. Basically 10:59:00 is rounded as 11:00:00 but 12:00:01 is rounded to 1day. Nevermind, just in case you decide to change this and round number to their nearest rounded value here is an alternative:

```c#
	public static TimeSpan Round(this TimeSpan ts, int minutes)
	{
		if (minutes > ts.TotalMinutes) return ts;
		var totalMinutes = ts.TotalMinutes + (minutes/2);
        return TimeSpan.FromMinutes(totalMinutes - (totalMinutes % minutes));
	}

	public static TimeSpan Round(this TimeSpan input) => input
		.Round(24 * 60 /* one day */)
		.Round(12 * 60 /* half day */)
		.Round(01 * 60 /* one hour */)
		.Round(01 * 30 /* half hour */);
```

This will fail to pass 3 tests that are very sensitive. Anyway, it would worth to review because if something is expecting to happen in one day and one minute then it sounds more logical to round that down to only one day than up to two days.

Reply:
**Round** is a bad name. I've changed it by **Reduce**, since it aligns more which what it does.

Its goal is to ignore less important parts of a time span. 
For instance:
- if the time span contains days, hours are rounded, so the resulting time span will be whole days.
- else if the time span contains hours, minutes are rounded, so the resulting time span will be whole hours
- else return the amount of minutes